ratelimiter/rsyslogstyle.go

```diff
@@ -0,0 +1,78 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/segmentio/rate-limiting-log-proxy/logger"
+)
+
+// RsyslogStyle is a rate limiter that works similar to rsyslog (and journald's)
+// rate limiters, allowing you to set an interval and a burst.  If there are more
+// messages than burst in a given interval, all further messages are dropped
+// until the end of that interval.
+type RsyslogStyle struct {
+	Interval time.Duration
+	Burst    int
+
+	start  time.Time
+	count  int
+	missed int
+	mu     sync.Mutex
+}
+
+// NewRsyslogStyle returns a RateLimiter based on rsyslog's rate limiter
+func NewRsyslogStyle(interval time.Duration, burst int) *RsyslogStyle {
+	return &RsyslogStyle{
+		Interval: interval,
+		Burst:    burst,
+	}
+}
+
+// Limit should be called each time you want to test if you've reached your rate
+// limit.  It will return true if you are over the limit and false otherwise.
+func (r *RsyslogStyle) Limit(logger logger.Logger) bool {
```

Comment:
Other rate limiters may want to log other information.  More importantly though, moving the logging logic into the handler means that either:
a) The handler needs to maintain state (per rate limiter) about when it got the first rejected limit, as well as the total count of rejected limits; or
b) The rate limiter needs to expose more of its internal state (expand the interface to allow checking if this was the first offense, number of dropped messages for an interval etc)

Given those choices, passing a logger seemed preferable.  Let me know if you had some other alternative in mind.